src/Microsoft.AspNetCore.Mvc.Core/AntiforgeryValidationFailedResult.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A <see cref="BadRequestResult"/> that is used for antiforgery
+    /// validation failures, and can be matched inside MVC result filters.
+    /// </summary>
+    public class AntiforgeryValidationFailedResult : BadRequestResult
```

Comment:
I'm not totally sure that subclassing `BadRequestResult` is the right thing to do here - this locks us in to a hierarchy we might not want to keep around forever. We tend to specify this kind of thing with and interface + subclass that ties the two together.

Basically I'd suggest keeping this class, and introducing a marker interface as well, then documenting that the **interface** is a stable contract.

Reply:
I like that idea! So I would create a `IAntiforgeryValidationFailedResult` then.

Since I don’t know exactly where what kind of types should be located, in what namespaces would I put the interface and the implementation then? Both in `Mvc`? I found some result interfaces inside `Mvc.Infrastructure` but those are probably meant for a somewhat internal use as well.

Reply:
Yeah, the interface probably belongs in `Mvc.Infrastructure`. The action result result belongs where the other action result types live.

Reply:
It feels a bit backwards since end users will probably never use the result object but rather the marker interface, but it matches how the other results and marker images are organized. Done! :slightly_smiling_face: